lib/dialects/abstract/connection-manager.js

```diff
@@ -31,10 +31,14 @@ class ConnectionManager {
       throw new Error('Support for pool:false was removed in v4.0');
     }
 
-    config.pool =_.defaults(config.pool || {}, defaultPoolingConfig, {
+    config.pool = _.defaults(config.pool || {}, defaultPoolingConfig, {
       validate: this._validate.bind(this),
-      Promise
-    }) ;
+      Promise: _.cloneDeep(Promise).config({
+        warnings: {
+          wForgottenReturn: false
```

Comment:
User can pass `Promise` from pool config, also we allow any type of `Promise` to be used here which may or may not support this config `wForgottenReturn`